### ISSUE TITLE
Extend Computer class from Player

### ISSUE THREAD

**Issue by ahthae** [DanielChvat/CIS-17B-Connect4Project#18]
So the Game doesn't have to treat computer turns as a special case, the Computer class should inherit the Player class and be passed to the Game.

**Comment by Computer-PsiPhi** [comment]
@ahthae where should the Computer object be created so that it can call the appropriate tkTurn() function?

**Comment by aethiu** [comment]
> @ahthae where should the Computer object be created so that it can call the appropriate tkTurn() function?

I'd have the Menu instantiate the players (computer or otherwise) by passing Player instances into the Game's constructor. We could even use the builder pattern (https://refactoring.guru/design-patterns/builder) to give us flexibility with how we construct the Game. It could have an interface like this: 
```c++
Menu::start_game() {
  GameBuilder builder;
  Game *game;
  
  builder.addPlayer(user1.player); // Adds User 1's player instance to the list of players.
  builder.addPlayer(new Computer()); // Adds a computer to the list of players.
  builder.setMode(5); // We could also use the builder to set the mode of the game, e.g. Connect 5.
  
  game = builder.build();
  game->run();
 }
```

**Comment by aethiu** [comment]
Fixed with #21
